elephant/spike_train_synchrony.py

```diff
@@ -341,33 +342,37 @@ def delete_synchrofacts(self, threshold, in_place=False, mode='delete'):
             if mode == 'extract':
                 mask = np.invert(mask)
             new_st = st[mask]
-            spiketrain_list[idx] = new_st
-            if in_place:
+            if in_place and st.segment is not None:
                 segment = st.segment
-                if segment is None:
-                    continue
 
-                # replace link to spiketrain in segment
-                new_index = self._get_spiketrain_index(
-                    segment.spiketrains, st)
-                segment.spiketrains[new_index] = new_st
+                try:
+                    # replace link to spiketrain in segment
+                    new_index = self._get_spiketrain_index(
+                        segment.spiketrains, st)
+                    segment.spiketrains[new_index] = new_st
+                except ValueError:
+                    # st is not in this segment even though it points to it
+                    warnings.warn(f"The SpikeTrain at index {idx} of the "
+                                  "input list spiketrains has a "
+                                  "unidirectional uplink to a segment in "
+                                  "whose segment.spiketrains list it does not "
+                                  "appear.")
```

Comment:
a typical elephant user might not digest such a message.
How about adding a description (on top of your sentence) about how to actually _fix_ the warning? Like "update the neo to the most recent version" or do/don't do things like `obj.segment = self.segment` manually.

Reply:
I have added more information on what's happening and on how to fix this. The warning is getting quite long though. What do you think?

Reply:
It's better even though it's longer.